pkg/cmd/infra/router/router.go

```diff
@@ -74,47 +131,92 @@ func NewCommandTemplateRouter(name string) *cobra.Command {
+	flag.StringVar(&cfg.F5RouterConfig.Password, "f5-password", util.Env("ROUTER_EXTERNAL_HOST_PASSWORD", ""), "The password for F5's management utility")
+	flag.StringVar(&cfg.F5RouterConfig.Vserver, "f5-vserver", util.Env("ROUTER_EXTERNAL_HOST_VSERVER", "ose-vserver"), "The F5 virtual server for OpenShift projects")
+	flag.BoolVar(&cfg.F5RouterConfig.Insecure, "f5-insecure", util.Env("ROUTER_EXTERNAL_HOST_INSECURE", "") == "true", "Skip strict certificate verification")
 
 	return cmd
 }
 
-func makeTemplatePlugin(cfg *templateRouterConfig) (*templateplugin.TemplatePlugin, error) {
-	if cfg.TemplateFile == "" {
+// isValidType checks that the router type specified is a known type
+func isValidType(t string) bool {
+	return t == routerTypeF5 || t == routerTypeTemplate
+}
+
+// makeTemplatePlugin creates a template router plugin
+func makeTemplatePlugin(cfg *routerConfig) (*templateplugin.TemplatePlugin, error) {
+	if cfg.TemplateRouterConfig.TemplateFile == "" {
 		return nil, errors.New("Template file must be specified")
 	}
 
-	if cfg.ReloadScript == "" {
+	if cfg.TemplateRouterConfig.ReloadScript == "" {
 		return nil, errors.New("Reload script must be specified")
 	}
 
 	statsPort := 0
 	var err error = nil
-	if cfg.StatsPort != "" {
-		statsPort, err = strconv.Atoi(cfg.StatsPort)
+	if cfg.TemplateRouterConfig.StatsPort != "" {
+		statsPort, err = strconv.Atoi(cfg.TemplateRouterConfig.StatsPort)
 		if err != nil {
 			return nil, errors.New("Invalid stats port")
 		}
 	}
 
 	templatePluginCfg := templateplugin.TemplatePluginConfig{
-		TemplatePath:       cfg.TemplateFile,
-		ReloadScriptPath:   cfg.ReloadScript,
-		DefaultCertificate: cfg.DefaultCertificate,
+		TemplatePath:       cfg.TemplateRouterConfig.TemplateFile,
+		ReloadScriptPath:   cfg.TemplateRouterConfig.ReloadScript,
+		DefaultCertificate: cfg.TemplateRouterConfig.DefaultCertificate,
 		StatsPort:          statsPort,
-		StatsUsername:      cfg.StatsUsername,
-		StatsPassword:      cfg.StatsPassword,
-		PeerService:        cfg.RouterService,
+		StatsUsername:      cfg.TemplateRouterConfig.StatsUsername,
+		StatsPassword:      cfg.TemplateRouterConfig.StatsPassword,
+		PeerService:        cfg.TemplateRouterConfig.RouterService,
 	}
 	return templateplugin.NewTemplatePlugin(templatePluginCfg)
 }
 
-// start launches the load balancer.
+// makeF5Plugin makes an F5 router plugin
+func makeF5Plugin(cfg *routerConfig) (*f5plugin.F5Plugin, error) {
+	if cfg.F5RouterConfig.Host == "" {
+		return nil, errors.New("F5 host must be specified")
+	}
+
+	if cfg.F5RouterConfig.Username == "" {
+		return nil, errors.New("F5 username must be specified")
+	}
+
+	if cfg.F5RouterConfig.Password == "" {
+		return nil, errors.New("F5 password must be specified")
+	}
+
+	if cfg.F5RouterConfig.Insecure == true {
+		glog.Warning("Strict certificate verification is *DISABLED*")
```

Comment:
I guess the principle is along the lines that validation of the parameters passed to the plug-in belongs here but additional logging output belongs in the plug-in itself? I'll move the warning to `newF5LTM`.